src/views/Home.jsx

```diff
@@ -1,22 +1,58 @@
 import './Home.css';
 import { useNavigate } from 'react-router-dom';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+import { db, findToken } from '../api/firebase';
+import { collection, getDocs, query } from 'firebase/firestore';
 
-export function Home({ handleClick, listToken }) {
+export function Home({ handleClick, listToken, setListToken }) {
 	const navigateTo = useNavigate();
+	const [joinListName, setJoinListName] = useState('');
+	const [error, setError] = useState(false);
 
 	useEffect(() => {
 		if (listToken) {
 			navigateTo('/list');
 		} else return;
 	}, [listToken, navigateTo]);
 
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setError(false);
+		const querySnapshot = await findToken(joinListName);
+		if (querySnapshot.size >= 1) {
```

Comment:
oooh I've never used .size before. Can I ask why you chose .size over .length?

Reply:
Oh yes! Great question! We tried to use .length, but the querySnapshot that is returned actually isn't an array! It's weird - it's a collection of objects, I think? The documentation says 'the number of documents can be determined via the `empty` and `size` properties.' (More in the docs [here](https://firebase.google.com/docs/reference/js/v8/firebase.firestore.QuerySnapshot) and [here](https://firebase.google.com/docs/reference/js/v8/firebase.firestore.QuerySnapshot#size).)

Reply:
Great question Mira! Thanks for the explanation Michelle!

Reply:
Thanks Mira and Lindsey. Mira asked basically my same question in a much more direct way. Thank you Michelle for the links to resources. I think I need to better familiarize myself with the firebase docs.